pgmpy/base/DAG.py

```diff
@@ -1583,6 +1583,139 @@ def edge_strength(self, data, edges=None):
 
         return strengths
 
+    def validate(self, data, metrics=None, **kwargs):
+
+        from sklearn.metrics import f1_score
+
+        from pgmpy.estimators.CITests import chi_square
+        from pgmpy.metrics.metrics import (
+            correlation_score,
+            fisher_c,
+            implied_cis,
+            permutation_test,
+            structure_score,
+        )
+
+        # all validation metrics
+        all_metrics = [
+            "correlation",
+            "aic",
+            "bic",
+            "fisher_c",
+            "implied_cis",
+            "permutation_test",
+        ]
+
+        # a normal validate call would provide all metric results
+        if metrics is None:
+            metrics = all_metrics
+
+        # Dictionary for holding default parameters
+        params = {
+            "correlation_test": "chi_square",
+            "correlation_significance": 0.05,
+            "correlation_score_func": f1_score,
+            "ci_test_permutation": chi_square,
+            "n_permutations": 1000,
+            "ci_test_fischer": chi_square,
+            "ci_test_cis": chi_square,
+            "ci_tests_significance": 0.5,
+            "calculate_rmsea": True,
+            "show_progress": True,
+        }
+
+        # For custom parameter updation
+        params.update(kwargs)
+
+        # to store the results of different tests
+        results = {}
+
+        # correlation score
+        if "correlation" in metrics:
+            results["Correlation Score"] = correlation_score(
+                model=self,
+                data=data,
+                test=params["correlation_test"],
+                significance_level=params["correlation_significance"],
+                score=params["correlation_score_func"],
+            )
+
+        # AIC-score with fail checks
+        if "aic" in metrics:
```

Comment:
Same, needs to be automatically detected. Check: https://github.com/pgmpy/pgmpy/blob/dev/pgmpy/estimators/StructureScore.py#L15